rust/src/lib/checkpoint.rs

```diff
@@ -0,0 +1,11 @@
+use crate::nm::{nm_checkpoint_destroy, nm_checkpoint_rollback};
+
+use crate::NmstateError;
+
+pub fn checkpoint_rollback(checkpoint: &str) -> Result<(), NmstateError> {
```

Comment:
Please place them as function of `NetworkState`. For example: `NetworkState::checkpoint_rollback()`.

Reply:
But this functions depends only on checkpoing not on NetworkState, they are free functions.